packages/perseus-editor/src/widgets/interactive-graph-editor/locked-figures/util.ts

```diff
@@ -124,3 +125,20 @@ export function generateLockedFigureAppearanceDescription(
             throw new UnreachableCaseError(fill);
     }
 }
+
+export async function generateSpokenMathDetails(mathString: string) {
+    const engine = await SpeechRuleEngine.setup("en");
+
+    // Replace everything between two $ signs with the spoken version
+    // of the math inside the $ signs.
+    // Example: "Circle with radius $\frac{1}{2}$" -> "Circle with radius one half"
+    const convertedSpeech = mathString.replace(
+        /\$(.*?)\$/g,
```

Comment:
This way of identifying TeX isn't going to work in all cases, because the TeX itself can contain dollar signs (either escaped, or in subexpressions).

For example: `This sandwich costs $\$50$.`

I might recommend using `@khanacademy/simple-markdown` (see perseus-markdown.tsx) to parse the string. You'll probably want to configure a separate parser instance in order to get TeX to render as speech. You could also use the parsing algorithm from `mathMatcher()`, defined in `pure-markdown/src/index.ts`